tachys/src/view/keyed/keyed_enumerate.rs

```diff
@@ -0,0 +1,445 @@
+use super::diff::{
+    diff, unpack_moves, Diff, DiffOpAdd, DiffOpAddMode, DiffOpMove,
+    DiffOpRemove, FxIndexSet, VecExt,
+};
+use crate::{
+    html::attribute::Attribute,
+    hydration::Cursor,
+    renderer::{CastFrom, Renderer},
+    ssr::StreamBuilder,
+    view::{
+        add_attr::AddAnyAttr, Mountable, Position, PositionState, Render,
+        RenderHtml,
+    },
+};
+use drain_filter_polyfill::VecExt as VecDrainFilterExt;
+use reactive_graph::{signal::ArcWriteSignal, traits::Set};
+use std::{hash::Hash, marker::PhantomData};
+
+/// Creates a keyed list of views.
+pub fn keyed_enumerate<T, I, K, KF, VF, V, Rndr>(
+    items: I,
+    key_fn: KF,
+    view_fn: VF,
+) -> KeyedEnumerate<T, I, K, KF, VF, V, Rndr>
+where
+    I: IntoIterator<Item = T>,
+    K: Eq + Hash + 'static,
+    KF: Fn(&T) -> K,
+    V: Render<Rndr>,
+    VF: Fn(usize, T) -> (ArcWriteSignal<usize>, V),
```

Comment:
This will probably fail to compile unless the `reactive_graph` feature is enabled on this crate. One of the goals is to abstract away the reactive system, and `tachys` as a whole doesn't depend on `reactive_graph`. This either needs to be feature-gated, or it needs to abstract over the concept of "set a signal" to something like "call this function if the index has changed"